src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -127,11 +127,11 @@ private static void AnalyzeSymbol(ISymbol symbol, SymbolAnalysisContext context)
             }
 
             var identifier = symbol.Name;
-            if (s_typeNames.Contains(identifier))
+            if (s_typeNames.Contains(identifier) && symbol.Locations.Length > 0)
```

Comment:
The exception occurs even before this is in the call `MatchesConfiguredVisibility` above. I have the fix with a unit test added in first commit of https://github.com/dotnet/roslyn-analyzers/pull/4066.